Sources/TokenGeneration/JWT/JWTGenerator.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+public struct JWTGenerator {
+    private let jwtRepresentation: JWTRepresentation
+    private let signingService: JWTSigningService
+    
+    public init(jwtRepresentation: JWTRepresentation,
+                signingService: JWTSigningService) {
+        self.jwtRepresentation = jwtRepresentation
+        self.signingService = signingService
+    }
+    
+    var token: String {
+        get throws {
+            guard let headerData = try? JSONSerialization.data(withJSONObject: jwtRepresentation.header),
```

Comment:
should we use `Codable` rather than `JSONSerialization` for encoding?

Reply:
![image 2024-11-19 at 16 31 47](https://github.com/user-attachments/assets/6761bc2f-a638-4682-b943-72f698fd01f9)

Reply:
Having the properties of this type serialised feels like it makes it extendable whereas having a type be decoded feels somewhat limited unless you've got a clever solution to that..? 🙂

Reply:
I think `any Encodable` rather than `Any` should fix that?

Reply:
Gotcha, to the second point, does this limit extendable use?
If we're happy to do that I can make the change but just want to make sure we're aligned 🙂

Reply:
Cool, as per Slack, let's stick with the `JSONSerialization` as `Encodable` seems to be a pain for heterogeneous dictionaries.